invokeai/app/services/shared/sqlite_migrator/migrations/migration_2.py

```diff
@@ -0,0 +1,196 @@
+        );
+        """,
+    ]
+
+    indices = [
+        "CREATE INDEX IF NOT EXISTS idx_workflow_library_created_at ON workflow_library(created_at);",
+        "CREATE INDEX IF NOT EXISTS idx_workflow_library_updated_at ON workflow_library(updated_at);",
+        "CREATE INDEX IF NOT EXISTS idx_workflow_library_opened_at ON workflow_library(opened_at);",
+        "CREATE INDEX IF NOT EXISTS idx_workflow_library_category ON workflow_library(category);",
+        "CREATE INDEX IF NOT EXISTS idx_workflow_library_name ON workflow_library(name);",
+        "CREATE INDEX IF NOT EXISTS idx_workflow_library_description ON workflow_library(description);",
+    ]
+
+    triggers = [
+        """--sql
+        CREATE TRIGGER IF NOT EXISTS tg_workflow_library_updated_at
+        AFTER UPDATE
+        ON workflow_library FOR EACH ROW
+        BEGIN
+            UPDATE workflow_library
+            SET updated_at = STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')
+            WHERE workflow_id = old.workflow_id;
+        END;
+        """
+    ]
+
+    for stmt in tables + indices + triggers:
+        cursor.execute(stmt)
+
+
+def _drop_model_manager_metadata(cursor: sqlite3.Cursor) -> None:
+    """Drops the `model_manager_metadata` table."""
+    cursor.execute("DROP TABLE IF EXISTS model_manager_metadata;")
+
+
+def _recreate_model_config(cursor: sqlite3.Cursor) -> None:
+    """
+    Drops the `model_config` table, recreating it.
+
+    In 3.4.0, this table used explicit columns but was changed to use json_extract 3.5.0.
+
+    Because this table is not used in production, we are able to simply drop it and recreate it.
+    """
+
+    cursor.execute("DROP TABLE IF EXISTS model_config;")
+
+    cursor.execute(
+        """--sql
+        CREATE TABLE IF NOT EXISTS model_config (
+            id TEXT NOT NULL PRIMARY KEY,
+            -- The next 3 fields are enums in python, unrestricted string here
+            base TEXT GENERATED ALWAYS as (json_extract(config, '$.base')) VIRTUAL NOT NULL,
+            type TEXT GENERATED ALWAYS as (json_extract(config, '$.type')) VIRTUAL NOT NULL,
+            name TEXT GENERATED ALWAYS as (json_extract(config, '$.name')) VIRTUAL NOT NULL,
+            path TEXT GENERATED ALWAYS as (json_extract(config, '$.path')) VIRTUAL NOT NULL,
+            format TEXT GENERATED ALWAYS as (json_extract(config, '$.format')) VIRTUAL NOT NULL,
+            original_hash TEXT, -- could be null
+            -- Serialized JSON representation of the whole config object,
+            -- which will contain additional fields from subclasses
+            config TEXT NOT NULL,
+            created_at DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+            -- Updated via trigger
+            updated_at DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+            -- unique constraint on combo of name, base and type
+            UNIQUE(name, base, type)
+        );
+        """
+    )
```

Comment:
I've updated this table to match these changes:
https://github.com/invoke-ai/InvokeAI/pull/5264#discussion_r1423406713

This migration represents changes from 3.4.0 to 3.5.0, so we can continue to update this migration file until v3.5.0 is released. We'll need a new migration file for subsequent changes